fix: gf fails on foo/bar.txt:line:column filepath (Windows) (#25236)
neovim/neovim: runtime/doc/news.txt

The following changes may require adaptations in user config or plugins.

> • colon (:) has been removed from |'isfname'| on Windows.

Reviewer: The news item should give a hint about the meaning of the change, e.g. "File path drive letters are now detected on Windows without needing to explicitly put ':' in 'isfname'." 

Also need to update `:h nvim-defaults`.

The following changes may require adaptations in user config or plugins.

> • File path drive letters are now detected on Windows without needing to
>   explicitly put ':' in 'isfname'.

Reviewer: ```suggestion
• Windows file path drive letters are now detected even if ":" is not in 'isfname'.
   And 'isfname' now omits ":" by default on Windows.
```
